### PR TITLE
Replace `Rose::Footer` with `Hds::AppFooter` component (HDS-2805)

### PR DESCRIPTION
**Pull request by KristinLBradley**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/HDS-2805)

## Description
If merged, this PR replaces the `Rose::Footer` component with the `Hds::AppFooter`. The `Rose::Footer` component will be removed in a separate follow-on PR after this PR is approved.

## Screenshots:
**BEFORE:**
image
image
image

---

**AFTER:**
image
image
image

**Dark Mode:**
(See [comment on HashiCorp logo color issue](https://github.com/hashicorp/boundary-ui/pull/1994#issuecomment-1810912655))
image

:technologist: [Admin preview](https://boundary-ui-git-hds-2805-adopt-hds-app-footer-hashicorp.vercel.app/scopes/global/authenticate/am_3enprm7y2t)

## Checklist:

- [x] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by KristinLBradley** [comment]
Currently in dark mode, the HashiCorp logo isn't visible. The `Hds::AppFooter` actually includes an argument to set either a "light" or "dark" theme though so I wonder if there's a way to switch to the "dark" theme when dark mode is enabled on Boundary... Otherwise, a style override could be added for the logo color in dark mode.

**Comment by DhariniJeeva** [comment]
@KristinLBradley, congrats on your first PR in Boundary! And thank you for putting this PR together so quickly. 👏 

To answer your question, we can access the current theme color. We can get the theme color from the session service. 

For example, the scopes [controller](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/controllers/scopes/scope.js) has the session service, and we can use the session to get the currently selected theme color.  Some of the other templates might not have a controller. In that case, we can create a new controller similar to [scopes](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/controllers/scopes/scope.js)
```
 <Hds::AppFooter @theme={{this.session.data.theme}} as |AF|>
      <AF.Link @href={{(doc-url)}}>{{t 'titles.documentation'}}</AF.Link>
      <AF.LegalLinks />
  </Hds::AppFooter>
```

Happy to work with you on this. Please let me know if you have any questions.

**Comment by alex-ju** [comment]
Looking good! @DhariniJeeva's suggestion sounds great – and I like how the values (`light`/`dark`) don't require any adjustments.

**Comment by calcaide** [comment]
Great work @KristinLBradley and as Dharini mentioned, thank you for your contribution!!

File: ui/admin/app/templates/scopes/index.hbs [hunk 1]
```diff
@@ -19,16 +18,12 @@
...
-          </nav.link>
-        </footer.nav>
-        <footer.brand @logo='logo-company' @text={{(company-name)}} />
-      </Rose::Footer>
+      <Hds::AppFooter @theme={{this.session.data.theme}} as |AF|>
+        <AF.Link
+          @href={{(doc-url)}}
+        >{{t 'titles.documentation'}}</AF.Link>
+        <AF.LegalLinks />
+      </Hds::AppFooter>
```
Comment (KristinLBradley) [inline ui/admin/app/templates/scopes/index.hbs:26]: @DhariniJeeva I wasn't able to figure out how to view this page in the browser. Could you tell me how to navigate to it? I think I need to add controller for it too but didn't want to add it blindly.

Comment (KristinLBradley) [reply ui/admin/app/templates/scopes/index.hbs:26]: Lisbet helped me find it so I've now updated it also. :)

**Comment by KristinLBradley** [comment]
Updates are ready to review. Please let me know if you have any feedback or if there is anything else that needs to be done.

**Review by calcaide — APPROVED** [review]
LGTM!! Awesome job @KristinLBradley thank you!

**Comment by DhariniJeeva** [comment]
How did we decide which links to add to the footer here? The component has more links, such as [status](https://hds-website-hashicorp.vercel.app/components/app-footer#status-link), but I'm not sure if we need that on the app. cc: @joshuaogle

**Review by DhariniJeeva — APPROVED** [review]

**Comment by joshuaogle** [comment]
This looks great! I think these standard links are good how they are now, status is just for the Cloud Platform. Thanks @KristinLBradley!

**Comment by KristinLBradley** [comment]
> How did we decide which links to add to the footer here? The component has more links, such as [status](https://hds-website-hashicorp.vercel.app/components/app-footer#status-link), but I'm not sure if we need that on the app. cc: @joshuaogle

The links currently included are the standardized "legal" links and the link to your documentation which you had previously.

**Pull request merged**